config/packages/dev/routing.yaml

```
@@ -1,3 +1,4 @@
 framework:
     router:
         strict_requirements: true
+        default_uri: 'http://localhost/'
```

You should set the actual URI in an environment variable (similar to `ROUTER_REQUEST_CONTEXT_HOST`), e.g.

```dotenv
DEFAULT_URI='http://localhost/'
```

```suggestion
        default_uri: '%env(DEFAULT_URI)%'
```

---

Does that matter if we have the URL in a `dev` config folder that only applies to dev?

---

It may not matter, but you assume that the sites runs on `http://localhost/` and that assumption may not be true. I never run projects on `http://localhost/`. Adding an environment variable for the URI makes it easy for developers to set the URI to whatever is the right value for their use.

---

The `.env` file is added to git so it wouldn't serve any purpose having the `default_uri` if it runs on the same `.env` file that the other environments run on. It would mean that switching the environment in the `.env` file and pushing it would change it for the other environments as well. If multiple developers work on the same project, we would have to add the `.env` file to `.gitignore` to allow each developer to run on the desired host they want, as well as handle that in the `docker-compose.yml` file or have a `docker-compose.override.yml` file. I will add it to the `.env` file if it helps the PR though

---

I have added the environment in the global `routing.yml` file since the environment variable key should be the same for all environment. That way all the environment's default url can be changed with the `DEFAULT_URI` environment variable

---

> The `.env` file is added to git so it wouldn't serve any purpose having the `default_uri` if it runs on the same `.env` file that the other environments run on. It would mean that switching the environment in the `.env` file and pushing it would change it for the other environments as well. If multiple developers work on the same project, we would have to add the `.env` file to `.gitignore` to allow each developer to run on the desired host they want, as well as handle that in the `docker-compose.yml` file or have a `docker-compose.override.yml` file. I will add it to the `.env` file if it helps the PR though

The `.env` file should contain sensible defaults and any local overrides should be made in `.env.local` (cf. https://symfony.com/doc/current/configuration.html#overriding-environment-values-via-env-local).